src/RpcServer/RpcServer.cs

```diff
@@ -129,11 +133,41 @@ public void StartRpcServer()
             }))
             .Configure(app =>
             {
+                if (settings.EnableCors)
+                    app.UseCors("All");
+
                 app.UseResponseCompression();
                 app.Run(ProcessAsync);
             })
             .ConfigureServices(services =>
             {
+                if (settings.EnableCors)
+                {
+                    if (settings.AllowOrigins.Length == 0)
+                        services.AddCors(options =>
+                        {
+                            options.AddPolicy("All", policy =>
+                            {
+                                policy.AllowAnyOrigin()
+                                .WithHeaders("Content-Type")
+                                .WithMethods("GET", "POST");
+                                // The CORS specification states that setting origins to "*" (all origins)
+                                // is invalid if the Access-Control-Allow-Credentials header is present.
+                            });
+                        });
+                    else
+                        services.AddCors(options =>
+                        {
+                            options.AddPolicy("All", policy =>
+                            {
+                                policy.WithOrigins(settings.AllowOrigins)
```

Comment:
I tried to specify a random origin policy `"AllowOrigins": ["http://yourapp.com"]`,automatically, I think I should had get an error while calling it from "localhost" but it kept responding.

Reply:
It doesn't care the origin only when you ask for it. Its a browser thing.

Reply:
In fact I forgot to update the client. It is blocking right now.